src/main.cpp

```diff
@@ -533,38 +508,48 @@ static bool read_config_file(const Settings &cmd_args)
 	return true;
 }
 
-static void init_debug_streams(int *log_level, const Settings &cmd_args)
+static void init_log_streams(const Settings &cmd_args)
 {
 #if RUN_IN_PLACE
-	std::string logfile = DEBUGFILE;
+	std::string log_filename = DEBUGFILE;
 #else
-	std::string logfile = porting::path_user + DIR_DELIM + DEBUGFILE;
+	std::string log_filename = porting::path_user + DIR_DELIM + DEBUGFILE;
 #endif
 	if (cmd_args.exists("logfile"))
-		logfile = cmd_args.get("logfile");
-
-	log_remove_output(&main_dstream_no_stderr_log_out);
-	*log_level = g_settings->getS32("debug_log_level");
-
-	if (*log_level == 0) //no logging
-		logfile = "";
-	if (*log_level < 0) {
-		dstream << "WARNING: Supplied debug_log_level < 0; Using 0" << std::endl;
-		*log_level = 0;
-	} else if (*log_level > LMT_NUM_VALUES) {
-		dstream << "WARNING: Supplied debug_log_level > " << LMT_NUM_VALUES
-		        << "; Using " << LMT_NUM_VALUES << std::endl;
-		*log_level = LMT_NUM_VALUES;
+		log_filename = cmd_args.get("logfile");
+
+	g_logger.removeOutput(&file_log_output);
+	std::string conf_loglev = g_settings->get("debug_log_level");
+
+	// Old integer format
+	if (std::isdigit(conf_loglev[0])) {
+		warningstream << "Deprecated use of debug_log_level with an "
+			"integer value; please update your configuration." << std::endl;
+		switch (atoi(conf_loglev.c_str())) {
+		case 0: return;  // No logging
+		case 1: conf_loglev = "error"; break;
+		case 2: conf_loglev = "action"; break;
+		case 3: conf_loglev = "info"; break;
+		case 4: conf_loglev = "verbose"; break;
+		default:
+			warningstream << "Supplied invalid debug_log_level!" << std::endl;
```

Comment:
Strange style. I don't think the body of the case should be on the same line. Even if that was acceptable then default should follow the same style. I also agree that case 0: return is pretty yuck (it's almost hidden).